server/service/users.go

```diff
@@ -844,10 +845,16 @@ func (svc *Service) PerformRequiredPasswordReset(ctx context.Context, password s
 	}
 
 	if user.SSOEnabled {
-		return nil, ctxerr.New(ctx, "password reset for single sign on user not allowed")
+		// should never happen because this would get caught by the
+		// CanPerformPasswordReset check above
+		err := fleet.NewPermissionError("password reset for single sign on user not allowed")
+		return nil, ctxerr.Wrap(ctx, err)
```

Comment:
sounds good, should we do the skip auth check call here too? just in case?

Reply:
It's not needed at this point because the authorize call has been done above (line 843).

Reply:
ah thank you, missed that